### PR TITLE
[productcatalogservice]add basic metric support

### PR DESCRIPTION
**Pull request by Frapschen**
# Changes

add basic metric support

image

related:https://github.com/open-telemetry/opentelemetry-demo/issues/66

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

### CODE REVIEW

File: src/productcatalogservice/main.go [hunk 1]
```diff
@@ -76,6 +92,18 @@ func main() {
 		}
 	}()
 
+	mp := initMeterProvider()
+	defer func() {
+		if err := mp.Shutdown(context.Background()); err != nil {
+			log.Printf("Error shutting down meter provider: %v", err)
```
Comment (fatsheep9146) [inline src/productcatalogservice/main.go:98]: printf -> fatalf

**Comment by julianocosta89** [comment]
> docs/services/productcatalogservice.md should also be updated.

https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/services/productcatalogservice.md#metrics

**Comment by Frapschen** [comment]
@fatsheep9146 @julianocosta89 please review :)

**Review by fatsheep9146 — APPROVED** [review]

**Review by julianocosta89 — APPROVED** [review]
I had to update the golang version and the `go get` command in order to be able to build the new container.
LGTM

**Pull request merged**